spec/jobs/bookmark_reminder_notifications_spec.rb

```diff
@@ -0,0 +1,144 @@
+    [
+      bookmark1,
+      bookmark2,
+      bookmark3
+    ]
+  end
+
+  before do
+    SiteSetting.enable_bookmarks_with_reminders = true
+    # this is done to avoid model validations on Bookmark
+    bookmark1.update_column(:reminder_at, five_minutes_ago - 10.minutes)
+    bookmark2.update_column(:reminder_at, five_minutes_ago - 5.minutes)
+    bookmark3.update_column(:reminder_at, five_minutes_ago)
+    Discourse.redis.flushall
+  end
+
+  it "sends every reminder and marks the reminder_at to nil for all bookmarks, as well as last sent date" do
+    subject.execute
+    bookmark1.reload
+    bookmark2.reload
+    bookmark3.reload
+    expect(bookmark1.reminder_at).to eq(nil)
+    expect(bookmark1.reminder_last_sent_at).not_to eq(nil)
+    expect(bookmark2.reminder_at).to eq(nil)
+    expect(bookmark2.reminder_last_sent_at).not_to eq(nil)
+    expect(bookmark3.reminder_at).to eq(nil)
+    expect(bookmark3.reminder_last_sent_at).not_to eq(nil)
+  end
+
+  it "will not send a reminder for a bookmark in the future" do
+    bookmark4 = Fabricate(:bookmark, reminder_at: Time.now.utc + 1.day)
+    BookmarkReminderNotificationHandler.expects(:send_notification).with(bookmark1)
+    BookmarkReminderNotificationHandler.expects(:send_notification).with(bookmark2)
+    BookmarkReminderNotificationHandler.expects(:send_notification).with(bookmark3)
+    BookmarkReminderNotificationHandler.expects(:send_notification).with(bookmark4).never
+    subject.execute
+    expect(bookmark4.reload.reminder_at).not_to eq(nil)
+  end
+
+  it "increments the job run number correctly and resets to 1 when it reaches 6" do
+    expect(Discourse.redis.get(described_class::JOB_RUN_NUMBER_KEY)).to eq(nil)
+    subject.execute
+    expect(Discourse.redis.get(described_class::JOB_RUN_NUMBER_KEY)).to eq('1')
+    subject.execute
+    subject.execute
+    subject.execute
+    subject.execute
+    subject.execute
+    expect(Discourse.redis.get(described_class::JOB_RUN_NUMBER_KEY)).to eq('6')
+    subject.execute
+    expect(Discourse.redis.get(described_class::JOB_RUN_NUMBER_KEY)).to eq('1')
+  end
+
+  context "when the bookmark with reminder site setting is disabled" do
+    it "does nothing" do
+      Bookmark.expects(:where).never
+      SiteSetting.enable_bookmarks_with_reminders = false
+      subject.execute
+    end
+  end
+
+  context "when one of the bookmark reminder types is at_desktop" do
+    let(:bookmark1) { Fabricate(:bookmark, reminder_type: :at_desktop) }
+    it "is not included in the run, because it is not time-based" do
+      BookmarkManager.any_instance.expects(:send_reminder_notification).with(bookmark1.id).never
+      subject.execute
+    end
+  end
+
+  context "when the number of notifications exceed MAX_REMINDER_NOTIFICATIONS_PER_RUN" do
+    it "does not send them in the current run, but will send them in the next" do
+      begin
+        original_const = Jobs::BookmarkReminderNotifications::MAX_REMINDER_NOTIFICATIONS_PER_RUN
```

Comment:
This is a bit awkward - I would prefer if we added a class variable for this and this code was changed to be a setter.

Reply:
So you mean changing this to @@max_reminder_notifications_per_run in the class and then passing in a different value via the execute method of the job (if neeeded, otherwise default like it does now to 300)?

Reply:
Yes, it could be changed to have a getter/setter and the test could call `Jobs::BookmarkReminderNotifications.max_reminder_notifications_per_run = 2`